### ISSUE TITLE
Typescript target files data is not included automatically

### ISSUE THREAD

**Issue by tsadigov** [istanbuljs/nyc#1095]
## Link to bug demonstration repository
https://github.com/tsadigov/nyc-ts

## Expected Behavior
when running nyc mocha  on test file X.spec.ts , I expect the tested files X.ts coverage to be computed

## Observed Behavior
the tests on X.spec.ts file are executed, a line is shown for the tested file in the report but it shows 0% which is wrong. I can get the right coverage report when adding the target file to the mocha list but that defies the purpose of setting up my environment. 

### Troubleshooting steps
npm test ./App.spec.ts
----------|----------|----------|----------|----------|-------------------|
File      |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
----------|----------|----------|----------|----------|-------------------|
All files |        0 |      100 |        0 |        0 |                   |
 App.ts   |        0 |      100 |        0 |        0 |         3,6,11,12 |
----------|----------|----------|----------|----------|-------------------|

npm test ./App.spec.ts ./App.ts
----------|----------|----------|----------|----------|-------------------|
File      |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
----------|----------|----------|----------|----------|-------------------|
All files |    61.54 |      100 |       25 |       60 |                   |
 App.ts   |    61.54 |      100 |       25 |       60 |         3,6,11,12 |
----------|----------|----------|----------|----------|-------------------|

## Environment Information

```
# paste the output here
  System:
    OS: Windows 10
    CPU: (4) x64 Intel(R) Xeon(R) CPU E31220 @ 3.10GHz
    Memory: 3.22 GB / 15.99 GB
  Binaries:
    Node: 10.15.2 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.13.0 - C:\Users\tural\AppData\Roaming\npm\yarn.CMD
    npm: 6.4.1 - C:\Program Files\nodejs\npm.CMD
  npmPackages:
    nyc: ^14.0.0 => 14.0.0
    ts-node: ^8.1.0 => 8.1.0
    typescript: ^3.4.5 => 3.4.5
```

**Comment by coreyfarrell** [comment]
Looks like you forgot to push to the demo repository.

**Comment by tsadigov** [comment]
git pushed it
sorry for that

**Comment by JaKXz** [comment]
I found a lot of configuration issues, if configured correctly the demo repo produces coverage reports correctly: https://github.com/tsadigov/nyc-ts/pull/1

**Comment by JaKXz** [comment]
I've updated the `@istanbuljs/nyc-config-typescript` readme: istanbuljs/istanbuljs#383

**Comment by tsadigov** [comment]
confirm that with updates you have made it compiles works.
I will try to understand the difference to apply to the main project. Guess the main diff is under devDependencies.

Before your commit I noticed that if the folder contains App.js file generated from App.ts it causes problem, so I used script to clear generated files before running nyc and excluded js files from config.

Using working configuration is the right way. Thanks a lot
